clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -1456,8 +1488,14 @@ public void handleServerDisconnect(long now, String destinationId, Optional<Auth
 
         @Override
         public void handleFailedRequest(long now, Optional<KafkaException> maybeFatalException) {
+            handleFailedRequest(now, maybeFatalException, updateMetadataLastRefreshTimeOnDisconnection);
```

Comment:
should we keep the default behavior as original:

handleFailedRequest(now, maybeFatalException, true);

Reply:
yeah, thanks for catching, meant to keep it true, got messed up somehow.